app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -52,16 +51,47 @@ def perform(form526_submission_id, full_name)
       end
     end
 
+    # class AccountNotFoundError < RuntimeError; end
+
     private
 
-    def bp_readings(client)
-      @bp_readings ||= client.list_resource('observations')
-      @bp_readings.present? ? FastTrack::HypertensionObservationData.new(@bp_readings).transform : []
-    end
+    class LighthouseVetsHealth
+      def initialize(form526_submission)
+        @form526_submission = form526_submission
+      end
+
+      def client
+        @client ||= Lighthouse::VeteransHealth::Client.new(account.icn.presence)
+      end
+
+      def account
+        user_uuid = @form526_submission.user_uuid.presence
+        edipi = @form526_submission.auth_headers['va_eauth_dodedipnid'].presence
+        account = Account.find_by(idme_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(logingov_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(edipi: edipi) if edipi
+        # Should we do this instead of expecting a NoMethodError?
+        # raise AccountNotFoundError.new("for user_uuid: #{user_uuid}") unless account&.icn
+        account
+      end
+
+      def bp_readings
+        @bp_readings ||= client.list_resource('observations')
+      end
 
-    def medications(client)
-      @medications ||= client.list_resource('medications')
-      @medications.present? ? FastTrack::HypertensionMedicationRequestData.new(@medications).transform : []
+      def observation_data
```

Comment:
Lighthouse uses this "observations" string but we're retrieving blood pressure readings specifically. I think this class bit easier to read when we are clear that we're looking at blood pressure readings and to avoid this "observations" key word that's really just something the client is passing on.

Reply:
I used `observation_data` because of the `HypertensionObservationData` class that is being used to `transform` the results. Since we are feeding in `bp_readings`, I'll rename the method -- suggestions?

Reply:
I'd probably call it "blood_pressure_readings" or "bp_readings".